astroquery/mast/cutouts.py

```diff
@@ -60,31 +72,50 @@ def _parse_cutout_size(size):
     if np.isscalar(size):
         size = np.repeat(size, 2)
 
+        limit_reached = size[0] > 30 or size[1] > 30
+
     if isinstance(size, u.Quantity):
         size = np.atleast_1d(size)
+
         if len(size) == 1:
             size = np.repeat(size, 2)
 
+        # Based on the literature, TESS resolution is approx. 21 arcseconds per pixel.
+        # We will convert the recommended upper limit for a dimension from pixels
+        # to the unit being passed.
+        unit = size[0].unit
+        upper_limit = (30 * 21*u.arcsec).to(unit).value
+        limit_reached = size[0].value > upper_limit or size[1].value > upper_limit
```

Comment:
The input `size` isn't always going to be in arcseconds, though. That's why `upper_limit` was created, so that the 30 pixel upper limit gets computed in whatever unit the user fed their cutout size with.

Reply:
being faster here also shouldn't be such a huge deal, hopefully this piece of code only runs once per query, and then after all we're querying a webservice, which would certainly dominate the overall speed.

Reply:
I agree performance isn't critical here. The suggested version is preferable mainly because it produces the same output with much less code. Being faster is just a nice bonus.

Reply:
The code is buggy as is, so I suggest first ensuring all acceptable input units are included in the test suite. Then I suppose there we'll need to have a both the upper limit definitions, but could also have your `any()` simplification.

Reply:
```suggestion
        with u.set_enabled_equivalencies(u.pixel_scale(21 * u.arcsec / u.pixel)):
            limit_reached = (size > 30 * u.pixel).any()
```

Reply:
Nice, I totally forgot about this equivalency.